File: src/EyeTimeTracker.Core/Sync/SyncSignatureBody.cs
namespace EyeTimeTracker.Core.Sync;

public static class SyncSignatureBody
{
    public static string ForSyncRequest(SyncRequest request)
    {
        return $"DeviceId={request.DeviceId ?? string.Empty}\n"
            + $"Platform={request.Platform ?? string.Empty}\n"
            + $"SinceUnixSeconds={request.SinceUnixSeconds}";
    }
}

File: android/EyeTimeTrackerAndroid/src/com/eyetimetracker/android/SyncSignatureBody.java
package com.eyetimetracker.android;

public final class SyncSignatureBody {
    private SyncSignatureBody() {
    }

    public static String forSyncRequest(String deviceId, String platform, long sinceUnixSeconds) {
        return "DeviceId=" + safe(deviceId)
                + "\nPlatform=" + safe(platform)
                + "\nSinceUnixSeconds=" + sinceUnixSeconds;
    }

    private static String safe(String value) {
        return value == null ? "" : value;
    }
}
